src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -678,22 +678,35 @@ export class TerminalInstance implements ITerminalInstance {
 
 	public preparePathForTerminalAsync(path: string): Promise<string> {
 		return new Promise<string>(c => {
-			const hasSpace = path.indexOf(' ') !== -1;
-			if (platform.isWindows) {
-				const exe = this.shellLaunchConfig.executable;
-				if (!exe) {
+			const exe = this.shellLaunchConfig.executable;
+			if (!exe) {
+				if (platform.isWindows) {
 					c(path);
-					return;
+				} else {
+					c(this._escapeNonWindowsPath(path));
 				}
+				return;
+			}
+
+			const hasSpace = path.indexOf(' ') !== -1;
+			const isPowerShell = exe.indexOf('pwsh') !== -1 ||
+				this.title.indexOf('pwsh') !== -1 ||
+				exe.toLocaleLowerCase().indexOf('powershell') !== -1 ||
+				this.title.toLocaleLowerCase().indexOf('powershell') !== -1;
```

Comment:
Same as above, should we check the title via ===?

Reply:
I think we need the `indexOf` here to cover the PowerShell extension which has the name "PowerShell Integrated Console". 

Granted, that's always started with either `powershell` or `pwsh` as the executable name so line 692 or 694 should already cover it...

I'm happy to make this change to `===` if you still think we should given this extra info.

Reply:
agreed and addressed!